.github/actions/tscommon/Makefile

```
@@ -0,0 +1,49 @@
+SHELL := /bin/bash
+ACTION_NAME = $(shell basename "$$(pwd)")
+export UNIT_TESTS_WD := $(shell echo "$$(pwd)")
+
+.PHONY: help
+help: ## Shows all targets and help from the Makefile (this message).
+	@echo "$(ACTION_NAME) Makefile"
+	@echo "Usage: make [COMMAND]"
+	@echo ""
+	@grep --no-filename -E '^([/a-z.A-Z0-9_%-]+:.*?|)##' $(MAKEFILE_LIST) | \
+		awk 'BEGIN {FS = "(:.*?|)## ?"}; { \
+			if (length($$1) > 0) { \
+				printf "  \033[36m%-20s\033[0m %s\n", $$1, $$2; \
+			} else { \
+				if (length($$2) > 0) { \
+					printf "%s\n", $$2; \
+				} \
+			} \
+		}'
+
+node_modules/.installed: package.json package-lock.json
+	npm ci
+	touch node_modules/.installed
+
+.PHONY: action
+action: node_modules/.installed ## Builds the action.
+	npm run build
+
+.PHONY: package
+package: node_modules/.installed ## Builds the distribution package.
+	npm run build
```

nit: I think this makes more sense.
```suggestion
	npm pack
```

You might also consider keeping the tgz in a dist/ directory and packing with `mkdir -p dist/ && npm pack --pack-destination dist/` kind of like how we have done with the other packages/actions. That's a bit more consistent.

---

we still need to build to "compile" TS to js. npm pack is only for the tarball, iiuc

---

made the changes, thatnks.

---

Right. you can have it depend on build.